oidc-api/src/main/java/uk/gov/di/authentication/oidc/lambda/AuthorisationHandler.java

```diff
@@ -180,7 +219,12 @@ public APIGatewayProxyResponseEvent authoriseRequestHandler(
             LOG.warn("No query string parameters are present in the Authentication request", e);
             throw new RuntimeException(
                     "No query string parameters are present in the Authentication request", e);
+        } catch (ClientNotFoundException e) {
+            LOG.warn("Invalid client or client not found in Client Registry");
+            return generateApiGatewayProxyResponse(
+                    400, OAuth2Error.INVALID_CLIENT.toJSONObject().toJSONString());
```

Comment:
Because this Lambda has been called via the browser instead of an API call from the frontend, the error response would need to be a 302 and contain the location. 

However as the client is invalid then as part of OAuth we should not return the user back to the RP but instead throw an exception. https://datatracker.ietf.org/doc/html/rfc6749#section-4.1.2.1